src/builtins.rs

```diff
@@ -1,6 +1,10 @@
+use std::collections::HashMap;
 use std::error::Error;
+use std::ptr::addr_of;
 
-const BUILTINS: &[&str] = &["cd", "pwd", "exit", "echo", "export"];
+static mut ALIASES: Option<HashMap<String, String>> = None;
```

Comment:
If we are doing alias substitution in `main.rs`, then it might be easier to just define it at the beginning there? Then you can pass it into `alias` and `expand_alias` and avoid all the unsafe code.